src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -919,7 +919,7 @@ private void checkUser()
             if (isFromType(ChannelType.PRIVATE))
                 throw new IllegalStateException("Cannot delete another User's messages in a PrivateChannel.");
 
-            IPermissionContainer permChan = (IPermissionContainer) channel;
+            IPermissionContainer permChan = getPermissionContainer();
```

Comment:
I think you can just directly use `getGuildChannel()` instead, or no?

Reply:
No, ends up with the same error.

Reply:
I do not get an error trying to do this:

```kt
val channel = event.channel as GuildChannel // in a thread
val bool = event.member!!.hasPermission(channel, Permission.MESSAGE_SEND)
println(bool) // true
```

Reply:
`hasPermission` accepts `GuildChannel` and knows to call `GuildChannel#getPermissionContainer()`